### PR TITLE
Allows syringes and droppers to show their reagents when examined while wearing things such as Science Goggles.

### LINKED ISSUE #19476: Cannot see reagents in a syringe/dropper
Syringes and droppers, when examined, do not print the reagents contained within if you are using an appropriate reagent scanner (such as Science Goggles).

This is because they are not considered open containers (which is appropriate). This is more of a feature request than a bug, I suppose. It would make sense that something that can see the reagents in a glass bottle should be able to see them in a syringe. Gameplay wise, it would helpful for chemist, xenobiologist, and virologist players.

### PR DESCRIPTION
**Pull request by Firecage**
This allows people to now see what reagents droppers and syringes contains when they are examined while wearing items with the reagent scanner functionality(such as Science Goggles). This is to fix a consistency issue(and it also didn't make any sense why you couldn't see what reagents they contained).

Fixes #19476

Edit: Once more tested now that I moved the defines from flags to container_type. Everything still appears to work as intended, and no runtimes either.

### CODE REVIEW

File: code/game/atoms.dm [hunk 1]
```diff
@@ -207,7 +210,7 @@
 	// *****RM
 	//user << "[name]: Dn:[density] dir:[dir] cont:[contents] icon:[icon] is:[icon_state] loc:[loc]"
 
-	if(reagents && is_open_container()) //is_open_container() isn't really the right proc for this, but w/e
+	if(reagents && is_open_container() || reagents && is_semi_open_container()) //is_open_container() isn't really the right proc for this, but w/e
```
Comment (GunHog) [inline code/game/atoms.dm]: Rearrange your logic here, you only need to check for reagents once.

Comment (Firecage) [inline code/game/atoms.dm]: @GunHog So basically

if(reagents && is_open_container() ||  is_semi_open_container()) ?

Comment (GunHog) [inline code/game/atoms.dm]: You MIGHT need to do `if(reagents && (is_open_container() || is_semi_open_container()))`. I do not remember the order of operations off the top of my head. Test both cases.

Comment (Firecage) [inline code/game/atoms.dm]: Sure, I'le do so. Thanks for the advice man.

File: code/__DEFINES/flags.dm [hunk 2]
```diff
@@ -24,6 +24,7 @@
 #define NOSLIP		1024 		//prevents from slipping on wet floors, in space etc (NOTE: flag shared with THICKMATERIAL for external suits and helmet)
 
 #define OPENCONTAINER	4096	// is an open container for chemistry purposes
+#define SEMIOPENCONTAINER	8192
```
Comment (xxalpha) [inline code/__DEFINES/flags.dm]: Just call it 'transparent'. "semiopen" isn't a realistic state.

Comment (Firecage) [inline code/__DEFINES/flags.dm]: @xxalpha >Realism in SS13.

Comment (xxalpha) [inline code/__DEFINES/flags.dm]: It's about logic and understanding, making the message clear. You're referring to glass containers here, they are not open, they are transparent, and that is the reason you can see their reagents.

Comment (Mervill) [inline code/__DEFINES/flags.dm]: and this is the _code_ not the game itself. The code must be as clear as possible (pun _intended_)

Comment (coiax) [inline code/__DEFINES/flags.dm]: why not just call the flag "REAGENTS_VISIBLE"

File: code/game/atoms.dm [hunk 3]
```diff
@@ -132,8 +132,8 @@
 /atom/proc/is_open_container()
 	return flags & OPENCONTAINER
 
-atom/proc/is_semi_open_container()
```
Comment (XDTM) [inline code/game/atoms.dm]: typo

also are you sure TRANSPARENT isn't defined elsewhere? could cause issues

File: code/game/atoms.dm [hunk 4]
```diff
@@ -132,6 +132,9 @@
 /atom/proc/is_open_container()
 	return flags & OPENCONTAINER
 
+atom/proc/is_transparet()
```
Comment (XDTM) [inline code/game/atoms.dm]: typo

also make sure TRANSPARENT isn't already defined elsewhere, could cause issues

Comment (Firecage) [inline code/game/atoms.dm]: And this is why I shouldn't type things when I am to sick to think straight.

File: code/game/atoms.dm [hunk 5]
```diff
@@ -132,6 +132,9 @@
 /atom/proc/is_open_container()
 	return flags & OPENCONTAINER
 
+atom/proc/is_transparent()
+	return flags & TRANSPARENT
+
```
Comment (coiax) [inline code/game/atoms.dm]: I feel like these procs can be turned into is_helpers DEFINES, which reduces our proc count. Because apparently in DM, proc calls are expensive. At least, that's what @RemieRichards tells me

Comment (Firecage) [inline code/game/atoms.dm]: Coiax, I am unsure how much me just changing these two procs would help, at all. Since I think we are at around 25k procs(the main procs and their children). Such a thing would probably have to be a whole PR of itself, focussing on more than just 1 or 2 procs.

Comment (xxalpha) [inline code/game/atoms.dm]: No, that would not be helpful. These things are better as procs because there might be conditions in which the atoms lose the TRANSPARENT flag and it's better to program those conditions in the is_transparent() proc.

File: code/__DEFINES/flags.dm [hunk 6]
```diff
@@ -24,6 +24,7 @@
 #define NOSLIP		1024 		//prevents from slipping on wet floors, in space etc (NOTE: flag shared with THICKMATERIAL for external suits and helmet)
 
 #define OPENCONTAINER	4096	// is an open container for chemistry purposes
+#define TRANSPARENT	8192
```
Comment (optimumtact) [inline code/__DEFINES/flags.dm]: This is already a define for flags?

#define BLOCK_GAS_SMOKE_EFFECT 8192	// blocks the effect that chemical clouds would have on a mob --glasses, mask and helmets ONLY! (NOTE: flag shared with THICKMATERIAL)
#define THICKMATERIAL 8192		//prevents syringes, parapens and hypos if the external suit or helmet (if targeting head) has this flag. Example: space suits, biosuit, bombsuits, thick suits that cover your body. (NOTE: flag shared with BLOCK_GAS_SMOKE_EFFECT)

Who is responsible for this shit?

Comment (Firecage) [inline code/__DEFINES/flags.dm]: @optimumtact I did test though if having a syringe/dropper would either make you immune from injections like hardsuits do, or if it would block gas smoke effects. Neither appears to have occurred.

Comment (coiax) [inline code/__DEFINES/flags.dm]: For less common flags, we could have a list of less common flags, normally set to null unless there's a less common flag, in which case it's checked for membership in the list, like species flags are. I think we could use @RemieRichards lazy lists as part of it.

We probably still need (I assume faster to check) binary number flags, but have some of the lesser stuff downgraded?

File: code/modules/research/rdmachines.dm [hunk 7]
```diff
@@ -47,7 +47,7 @@
 
 
 
-/obj/machinery/r_n_d/attackby(obj/item/O, mob/user, params)
+/obj/machinery/r_n_d/attackby(obj/O, mob/user, params)
```
Comment (optimumtact) [inline code/modules/research/rdmachines.dm]: why did you change the path of this?

**Comment by optimumtact** [comment]
❤️

File: code/modules/research/rdmachines.dm (hunk 7 above)
Comment (Firecage) [inline code/modules/research/rdmachines.dm]: Ah right, that was when I was originally busy changing the defines the first time, forgot to change them back.

**Comment by optimumtact** [comment]
One last thing, can you move the defines to either reagents or make a new container_type.dm file

**Comment by Firecage** [comment]
Ah, so that .dm file is less cluttered? Would probably be the most consistency in the reagents one then.

**Comment by optimumtact** [comment]
Well yeah, and they're not technically defines for the flag var itself now 😁

**Comment by optimumtact** [comment]
👍 will merge when travis passes

**Comment by Firecage** [comment]
@optimumtact And thus Travis has spoken!
